amoro-format-iceberg/src/main/java/org/apache/amoro/optimizing/plan/AbstractOptimizingPlanner.java

```diff
@@ -158,19 +158,23 @@ public List<RewriteStageTask> planTasks() {
     double maxInputSize = maxInputSizePerThread * availableCore;
     actualPartitionPlans = Lists.newArrayList();
     long actualInputSize = 0;
+    double avgThreadCost = actualInputSize / availableCore;
+    List<RewriteStageTask> tasks = Lists.newArrayList();
+
     for (PartitionEvaluator evaluator : evaluators) {
-      actualPartitionPlans.add((AbstractPartitionPlan) evaluator);
-      actualInputSize += evaluator.getCost();
+      AbstractPartitionPlan actualPartitionPlan = (AbstractPartitionPlan) evaluator;
+      List<RewriteStageTask> splitTasks =
+          actualPartitionPlan.splitTasks((int) (actualInputSize / avgThreadCost));
```

Comment:
It is ok to change this currently, because the parameter for `splitTasks` is not taken into consideration, but maybe in the future we will change the logic in `TaskSplitter#splitTasks`

Can we change the `actualInputSize / avgThreadCost` here to some other things like `evaluator.getCost() / avgThreadCostPre` -- but currently there is no `avgThreadCostPre` 

Maybe we can add a comment here to let others know we need to change this here? what do you think about this

Reply:
I see what you mean, will publish another review.

Regarding the comment I think it should be placed in the implementation classes.